cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -182,6 +183,63 @@ def validate_accelerator_requirements(
         self.validate_accelerator_count(compute_resource_name, instance_types_info)
         self.validate_accelerator_manufactures(compute_resource_name, instance_types_info)
 
+    def validate_efa_requirements(
+        self,
+        compute_resource_name: str,
+        instance_types_info: Dict[str, InstanceTypeInfo],
+        efa_enabled: bool,
+    ):
+        """Check if the instance types have a uniform support for EFA.
+
+        Validation Failure is expected if EFA is ENABLED and at least one instance type defined in the compute resource
+        DOES NOT support EFA.
+        """
+        if efa_enabled:
+            all_instance_types = set(instance_types_info.keys())
+            instance_types_without_efa_support = {
+                instance_type_name
+                for instance_type_name, instance_type_info in instance_types_info.items()
+                if not instance_type_info.is_efa_supported()
+            }
+
+            # If all the instance types have EFA support, `instance_types_without_efa_support` should be empty
+            # --> No failure expected
+            # If all the instance types DO NOT have EFA support, `instance_types_without_efa_support` should be the same
```

Comment:
> If all the instance types DO NOT have EFA support, `instance_types_without_efa_support` should be the same as `all_instance_types` (Set difference of 0) --> No failure expected

I think this should be a failure because EFA is enabled on LT and NO instance will be launched, since none of them support EFA